### PR TITLE
PCHR-2285: Open request details on listed request click

### PR DESCRIPTION
**Pull request by igorpavlov**
⚠️ This branch must wait for sync from https://github.com/civicrm/civihr/pull/2039 and then be merged immediately

## Overview

This PR adds an ability to view/respond/edit a leave request from Manager Leave or Admin Dashboard Leave Requests pages by simply clicking on a request *row* in the table.

## Technical Details

1. The table now has a `.table-hover` class (according to the Bootstrap StyleGuide)
2. Each row has a `.pointer` class
3. Actions (dropdown) component has instructions to stop event propagation on click events (do not fire clicks on parent component, i.e. table row).

### Refactoring

This PR has a refactoring of Leave Request directive - it now utilises session and does **not** need `contact-id` to be passed.

Example 1 - When creating a request
```html
<!-- Do not pass contact-id to the directive anymore -->
<a href
  leave-request-popup
  leave-type="toil"
  selected-contact-id="203"
  is-self-record="true">
  Create new request
</a>
```
Example 2 - When editing a request
```html
<!-- Do not pass contact-id to the directive anymore -->
<a href
  leave-request-popup
  leave-request="request"
  is-self-record="false">
  Edit request
</a>
```
 `contact-id` passing was removed from several view templates.

### Session Model Support

Session is refactored in https://github.com/civicrm/civihr/pull/2039

Session is used in 2 files now (the .spec one uses a mock).

image

Since Session is now a Model, the file paths and dependencies were changed accordingly.

## Comments

1. _Note: any view templates connected with calendars has **not** been yet touched due to the potential clashing with the following PR (https://github.com/civicrm/civihr/pull/1989). `contact-id` passing should be removed later on._
2. Tests are amended to keep the same logic and checks, but without considering `contact-id` to be passed.
3. JS styling improvements were done.

---

- [x] Tests Pass

### CODE REVIEW

**Comment by deb1990** [comment]
> his PR has a refactoring of Leave Request directive - it now utilises session and does not need contact-id to be passed.
> Example 1 & 2

The above should be moved to either Technical Discussion's Refactoring sub section

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js [hunk 1]
```diff
@@ -525,7 +527,10 @@ define([
         initAvailableStatusesMatrix.call(this);
         initListeners.call(this);
 
-        return initRoles.call(this)
+        return loadLoggedInContactId.call(this)
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: Extra function can be removed
```
return loadLoggedInContactId.call(this)
.then(initRoles.bind(this))
.then(this._initRequest.bind(this))
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ define([
   'common/moment',
   'common/services/api/option-group',
   'common/services/hr-settings',
+  'common/services/session',
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: This file had lot of violations of the angular style guide. As we touched it, should be fixed.

Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: Example: 
1. Order of injection for both requirejs and angular
2. Function Declarations to Hide Implementation Details
etc

Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: Please do it in a separate commit

**Comment by deb1990** [comment]
1. The hover effect is not visible in the screencast, are you sure it is working?
2. The whole fix is also working for the SSP Manager leave page right?

**Comment by igorpavlov** [comment]
@deb1990 

> The hover effect is not visible in the screencast, are you sure it is working?

Do you mean changing colour? If yes, then it is visible on the After GIF.

> The whole fix is also working for the SSP Manager leave page right?

Correct. As it is stated in the overview, the enhancement (not a fix actually) works on both Manager Leave and Admin Dashboard Leave Requests pages

**Comment by deb1990** [comment]
@igorpavlov  Yes I was checking on the before GIF. It looks good.

**Comment by igorpavlov** [comment]
Also removed `is-self-record` parameter from the directive call for rows, since it is always false in this case.
